test/framework/src/main/java/org/elasticsearch/index/mapper/FieldTypeTestCase.java

```diff
@@ -47,6 +47,17 @@ private static SearchExecutionContext createMockSearchExecutionContext(boolean a
         ValueFetcher fetcher = fieldType.valueFetcher(searchExecutionContext, format);
         SourceLookup lookup = new SourceLookup();
         lookup.setSource(Collections.singletonMap(field, sourceValue));
-        return fetcher.fetchValues(lookup, Collections.emptySet());
+        return fetcher.fetchValues(lookup);
+    }
+
+    public static List<?> fetchSourceValues(MappedFieldType fieldType, Object... values) throws IOException {
```

Comment:
One motivation for this PR is to return well-formed values for a document field, even if its some of its values were ignored. So it'd be good to add a test for this case explicitly.

Reply:
I've added a yaml test for this